File: server/utils/types/commanders.py
from typing import Optional
from pydantic import BaseModel


class Commander(BaseModel):
    colorID: Optional[str] = None
    wins: Optional[int] = None
    winsSwiss: Optional[int] = None
    winsBracket: Optional[int] = None
    draws: Optional[int] = None
    losses: Optional[int] = None
    lossesSwiss: Optional[int] = None
    lossesBracket: Optional[int] = None
    count: Optional[int] = None
    winRate: Optional[float] = None
    winRateSwiss: Optional[float] = None
    winRateBracket: Optional[float] = None
    topCuts: Optional[int] = None
    conversionRate: Optional[float] = None

    class Config:
        extra = "forbid"


class Commanders(BaseModel):
    commanders: dict[str, Commander]

    class Config:
        extra = "forbid"

File: client/src/utils/types/commanders.ts
import { z } from "zod";

export const Commander = z.object({
  colorID: z.string().nullish(),
  wins: z.number().nullish(),
  winsSwiss: z.number().nullish(),
  winsBracket: z.number().nullish(),
  draws: z.number().nullish(),
  losses: z.number().nullish(),
  lossesSwiss: z.number().nullish(),
  lossesBracket: z.number().nullish(),
  count: z.number().nullish(),
  winRate: z.number().nullish(),
  winRateSwiss: z.number().nullish(),
  winRateBracket: z.number().nullish(),
  topCuts: z.number().nullish(),
  conversionRate: z.number().nullish(),
  colors: z.unknown().nullish(),
});

export const Commanders = z.record(Commander);

export type CommanderType = z.infer<typeof Commander>;
export type CommandersType = z.infer<typeof Commanders>;
